apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1323 @@
+		PRNT("Error: Incorrect method value!!");
+		goto ret_error;
+	}
+
+	if (request->encoding != CONTENT_LENGTH && request->encoding != CHUNKED_ENCODING) {
+		PRNT("Error: Incorrect encoding value!!");
+		goto ret_error;
+	}
+
+	request->buffer = (char *)zalloc(request->buflen);
+	if (request->buffer == NULL) {
+		PRNT("Error: Fail to allocate buffer");
+		goto ret_error;
+	}
+
+	if (request->buflen <= 0) {
+		PRNT("Error: Buffer length must be bigger than 0!!");
+		return WS_TEST_ERR;
+	}
+
+	request->tls = false;
+	request->response = response;
+	request->callback = cb;
+
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (ssl_conf) {
+		request->tls = true;
+		memcpy(&request->ssl_config, ssl_conf, sizeof(struct http_client_ssl_config_t));
+
+		data = mallinfo();
+		if (data.fordblks < WS_TEST_CONF_MIN_TLS_MEMORY) {
+			PRNT("Error: Not enough memory!!");
+			goto ret_error;
+		}
+	}
+#endif
+
+	res = ws_test_base((void *)request);
+	if (res != 0) {
+		PRNT("ws_test_base() failed %d", res);
+	}
+
+	free(request->buffer);
+	FUNC_EX;
+	return res;
+
+ret_error:
+	FUNC_EX;
+	return WS_TEST_ERR;
+}
+
+static int ws_test_prepare_request(int argc, char** argv, struct http_client_request_t *request)
+{
+	FUNC_EN;
+	char *p, *q;
+	g_is_test_entity = 0;
+	g_keep_alive_count = 0;
+	g_chunk_count = 0;
+	g_trailer = 0;
+	memset(request, 0, sizeof(struct http_client_request_t));
+
+
+	if (!strncmp(argv[2], "GET", 4)) {
+		request->method = WS_TEST_MODE_GET;
+	} else if (!strncmp(argv[2], "PUT", 4)) {
+		request->method = WS_TEST_MODE_PUT;
+	} else if (!strncmp(argv[2], "POST", 5)) {
+		request->method = WS_TEST_MODE_POST;
+	} else if (!strncmp(argv[2], "DELETE", 7)) {
+		request->method = WS_TEST_MODE_DELETE;
+	} else {
+		PRNT("invalid req_type");
+		return WS_TEST_ERR;
+	}
+
+	request->url = argv[3];
+
+#ifdef CONFIG_NET_SECURITY_TLS
+	if (!strncmp(request->url, "https", 5)) {
+		g_https = 1;
+	} else
+#endif
+	if (!strncmp(request->url, "http", 4)) {
+		g_https = 0;
+	} else {
+		PRNT("issue with url");
+		return WS_TEST_ERR;
+	}
+
+	for (int i = 4; i < argc; i++) {
+		p = argv[i];
+		if ((q = strchr(p, '=')) == NULL) {
+			PRNT("issue in optional args, '=' not found");
+			return WS_TEST_ERR;
+		}
+		q++;
+
+		if (strncmp(p, "test_entity", 11) == 0) {
+			int t = atoi(q);
+			if (t > 0 && t <= WS_TEST_CONF_MAX_ENTITY_SIZE) {
+				request->entity = (char *)malloc(t+1);
+				if (request->entity == NULL) {
+					PRNT("no memory allocated");
+					return WS_TEST_ERR;
+				}
+				g_is_test_entity = 1;
+				memset(request->entity, '1', t);
+				request->entity[t] = '\0';
+			} else {
+				PRNT("%d is out of range",t);
+				return WS_TEST_ERR;
+			}
+		}
+		else  if (strncmp(p, "keep-alive", 10) == 0) {
+			g_keep_alive_count = atoi(q);
+		}
+		else if (strncmp(p, "chunk-count", 11) == 0) {
+			request->encoding = 1;
+			g_chunk_count = atoi(q);
+		}
+		else if (strncmp(p, "trailer", 7) == 0) {
+			g_trailer = atoi(q);
```

Comment:
please minimize global variable and add g_trailer in structure http_client_request_t